### PR TITLE
Dev spring boot book tests

### PR DESCRIPTION
**Pull request by OlegTuryk**

### CODE REVIEW

File: src/test/java/com/app/onlinebookstore/controller/BookControllerTest.java [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+        mockMvc = MockMvcBuilders
+                .webAppContextSetup(applicationContext)
+                .apply(springSecurity())
+                .build();
+
+        CATEGORY.setId(1L);
+        CATEGORY.setName("Test Category 1");
+        CATEGORY.setDescription("Description for Test Category 1");
+
+        CREATE_BOOK_REQUEST_DTO.setTitle("Test Title1");
```
Comment (andrii-hoienko) [inline src/test/java/com/app/onlinebookstore/controller/BookControllerTest.java:63]: Please create constants wich fields will be populated and reuse them in test methods

File: src/test/java/com/app/onlinebookstore/controller/BookControllerTest.java [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+        MvcResult result = mockMvc.perform(
+                        post("/books")
+                                .content(jsonRequest)
+                                .contentType(MediaType.APPLICATION_JSON)
+                )
+                .andExpect(status().isOk())
+                .andReturn();
+        BookDto actual = objectMapper.readValue(result
+                .getResponse().getContentAsString(), BookDto.class);
+        Assertions.assertNotNull(actual);
```
Comment (andrii-hoienko) [inline src/test/java/com/app/onlinebookstore/controller/BookControllerTest.java]: please use static import

File: src/test/java/com/app/onlinebookstore/controller/BookControllerTest.java [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+        Assertions.assertNotNull(actual.getId());
+        assertTrue(EqualsBuilder.reflectionEquals(bookDto, actual, "id"));
```
Comment (andrii-hoienko) [inline src/test/java/com/app/onlinebookstore/controller/BookControllerTest.java:115]: What is the reason to use 'assertTrue(EqualsBuilder.reflectionEquals(...' is it possible to assertEquals?

File: src/test/java/com/app/onlinebookstore/service/BookServiceTest.java [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+
+    @Test
+    @DisplayName("Save book - Valid book")
+    public void save_ValidBook_ReturnSavedBookDto() {
+        CreateBookRequestDto createBookRequestDto = new CreateBookRequestDto();
+        createBookRequestDto.setTitle("Test Title");
+        createBookRequestDto.setAuthor("Test Author");
+        createBookRequestDto.setIsbn("978-3-16-148410-0");
+        createBookRequestDto.setPrice(new BigDecimal("19.99"));
+        when(bookMapper.toModel(createBookRequestDto)).thenReturn(book);
```
Comment (andrii-hoienko) [inline src/test/java/com/app/onlinebookstore/service/BookServiceTest.java:90]: To make test more readable separate with empty lines precondition, action and assertion

File: src/test/java/com/app/onlinebookstore/service/BookServiceTest.java [hunk 5]
```diff
@@ -0,0 +1,182 @@
...
+        assertNotNull(actual);
+        assertEquals(bookDto, actual);
+    }
+
+    @Test
+    @DisplayName("Get book by id - Book not found")
+    public void getById_BookNotFound_ThrowEntityNotFoundException() {
+        Long nonExistentId = 2L;
+        when(bookRepository.findByIdWithCategories(nonExistentId)).thenReturn(Optional.empty());
+        assertThrows(EntityNotFoundException.class, () -> bookService.getById(nonExistentId));
```
Comment (andrii-hoienko) [inline src/test/java/com/app/onlinebookstore/service/BookServiceTest.java]: Check error message as well

File: src/test/java/com/app/onlinebookstore/service/BookServiceTest.java [hunk 6]
```diff
@@ -0,0 +1,182 @@
...
+    }
+
+    @Test
+    @DisplayName("Delete book by id")
+    public void deleteById_BookExists_DeleteSuccessfully() {
+        doNothing().when(bookRepository).deleteById(TEST_ID);
+        bookService.deleteById(TEST_ID);
+        verify(bookRepository, times(1)).deleteById(TEST_ID);
```
Comment (andrii-hoienko) [inline src/test/java/com/app/onlinebookstore/service/BookServiceTest.java]: times(1) set by default

File: src/test/resources/database/delete/delete-book-and-category-data.sql [hunk 7]
```diff
@@ -0,0 +1,5 @@
+DELETE FROM book_category;
+
+DELETE FROM books;
+
+DELETE FROM categories;
```
Comment (andrii-hoienko) [inline src/test/resources/database/delete/delete-book-and-category-data.sql]: empty line

**Review by andrii-hoienko — CHANGES_REQUESTED** [review]
GJ! But still, some improvements required

**Review by vovasalyha — APPROVED** [review]

**Pull request merged**